### PR TITLE
Add new sync-policy related params to boto3 extension

### PR DESCRIPTION
**Pull request by soumyakoduri**
## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [x] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

### CODE REVIEW

File: examples/rgw/boto3/service-2.sdk-extras.json [hunk 1]
```diff
@@ -46,6 +46,37 @@
...
+	    }
+	},
+	"S3RepSource": {
+	    "type": "structure",
+	    "members": {
+                "Zone": {
+                    "shape":"Zone",
+                    "documentation":"<p>Replication source zone.</p>",
+                    "location":"querystring",
+                    "locationName":"Zone"
```
Comment (cbodley) [inline examples/rgw/boto3/service-2.sdk-extras.json:65]: these aren't in the query string, so we don't need the "location". and i think "locationName" can be omitted if it matches the name of the member

**Review by cbodley — COMMENTED** [review]
very cool

**Comment by cbodley** [comment]
i see from https://docs.aws.amazon.com/AmazonS3/latest/API/API_ReplicationRule.html#AmazonS3-Type-ReplicationRule-SourceSelectionCriteria that aws supports a \<SourceSelectionCriteria\> under \<ReplicationRule\>

we might want to add support for our \<Zone\> extension there too

File: examples/rgw/boto3/service-2.sdk-extras.json [hunk 2]
```diff
@@ -46,6 +46,37 @@
                 }
             }
         },
+        "ReplicationRule":{
+            "members":{
+		"Source": {
+		    "shape":"S3RepSource",
+		    "documentation":"Replication source zone.",
```
[the remark this answers is not in the document]
Comment (soumyakoduri) [reply examples/rgw/boto3/service-2.sdk-extras.json:53]: Done.

File: examples/rgw/boto3/service-2.sdk-extras.json [hunk 3]
```diff
@@ -46,6 +46,37 @@
...
+		    "locationName":"Source"
+		}
+	    }
+	},
+	"S3RepSource": {
+	    "type": "structure",
+	    "members": {
+                "Zone": {
+                    "shape":"Zone",
+                    "documentation":"<p>Replication source zone.</p>",
```
[the remark this answers is not in the document]
Comment (soumyakoduri) [reply examples/rgw/boto3/service-2.sdk-extras.json:63]: Done.

File: examples/rgw/boto3/service-2.sdk-extras.json (hunk 1 above)
Comment (soumyakoduri) [reply examples/rgw/boto3/service-2.sdk-extras.json:65]: Thanks..Done.

**Comment by cbodley** [comment]
> ok.. But RGW doesn't support yet (https://bugzilla.redhat.com/show_bug.cgi?id=2272030#c1)

it's outside the scope of this pr, but it would be simple enough to support our \<Zone\> extension under \<SourceSelectionCriteria\> instead of \<Source\> without adding other elements like SseKmsEncryptedObjects/ReplicaModifications

**Comment by soumyakoduri** [comment]
> > ok.. But RGW doesn't support yet (https://bugzilla.redhat.com/show_bug.cgi?id=2272030#c1)
> 
> it's outside the scope of this pr, but it would be simple enough to support our <Zone> extension under <SourceSelectionCriteria> instead of <Source> without adding other elements like SseKmsEncryptedObjects/ReplicaModifications

Ah got it! 
So it is fine to modify the existing params' placement at any time .. is it?

**Comment by cbodley** [comment]
> Ah got it! So it is fine to modify the existing params' placement at any time .. is it?

i assume we'd continue to parse \<Source\> too if there's interest in maintaining backward compatibility

**Review by cbodley — APPROVED** [review]

**Comment by cbodley** [comment]
https://jenkins.ceph.com/job/ceph-windows-pull-requests/40092/
> [453/1072] Building CXX object src/mgr/CMakeFiles/ceph-mgr.dir/DaemonKey.cc.o
client_loop: send disconnect: Broken pipe

**Comment by cbodley** [comment]
jenkins test windows

**Pull request merged**